bokehjs/src/coffee/document.coffee

```diff
@@ -650,8 +654,13 @@ export class Document
             throw new Error("Cannot apply patch to #{patched_id} which is not in the document")
           patched_obj = @_all_models[patched_id]
           attr = event_json['attr']
-          value = Document._resolve_refs(event_json['new'], old_references, new_references)
-          patched_obj.setv({ "#{attr}" : value })
+          model_type = event_json['model']['type']
+          if attr == 'data' and model_type == 'ColumnDataSource'
+            [data, shapes] = decode_column_data(event_json['new'])
+            patched_obj.setv({_shapes: shapes, data: data}, {setter_id: setter_id})
```

Comment:
this makes me think, we aren't handling _resolve_refs in this branch, which means we don't allow column data to contain models; which is probably a good idea; but it isn't captured by `Dict(String, Seq(Any))` in the Python model.
Is this the only way in which that `Any` is a lie or are there tons of other unacceptable column types?